src/rustup-cli/self_update.rs

```diff
@@ -1069,13 +1070,22 @@ pub fn update() -> Result<()> {
     }
     let setup_path = try!(prepare_update());
     if let Some(ref p) = setup_path {
-        info!("rustup updated successfully");
+        let version = try!(get_new_rustup_version(&p));
+        info!("rustup updated successfully to {}", version);
         try!(run_update(p));
     }
 
     Ok(())
 }
 
+fn get_new_rustup_version(path: &Path) -> Result<String> {
+    let raw_version = Command::new(path).arg("--version").output().unwrap().stdout;
+    let string_version = String::from_utf8(raw_version).unwrap_or(String::new());
+    let re = Regex::new(r"\d+.\d+.\d+").unwrap();
+    let capture = re.captures(&string_version).unwrap().at(0).unwrap();
+    Ok(String::from(capture))
+}
```

Comment:
Yes absolutely! I'll update the code to be more defensive. Do you feel like regexp is the way to go or could a string split be sufficient?